packages/calcite-components/src/components/tile/tile.scss

```diff
@@ -70,7 +93,7 @@
 }
 
 .heading {
-  color: var(--calcite-tile-heading-text-color);
+  color: var(--calcite-tile-heading-text-color, var(--calcite-internal-tile-heading-text-color));
```

Comment:
heading and description do not have unique styles. Instead by default they share a text color which does not changes depending on state. We do not want to break our current design pattern here. Therefore, both heading and description should use a shared `--calcite-tile-text-color` with a fallback to "currentColor"